pkg/cache/v3/resource.go

```diff
@@ -97,6 +97,23 @@ func GetResourceName(res types.Resource) string {
 	}
 }
 
+// ResourceRequiresFullStateInSotw indicates whether when building the reply in Sotw,
+// the response must include all existing resources or can return only the modified ones
+func ResourceRequiresFullStateInSotw(typeURL resource.Type) bool {
+	// From https://www.envoyproxy.io/docs/envoy/v1.28.0/api-docs/xds_protocol#grouping-resources-into-responses,
+	// when using sotw the control-plane MUST return all requested resources (or simply all if wildcard)
+	// for some types. This is relied on by xds-grpc which is explicitly requesting clusters but expect
+	// to receive all existing resources
```

Comment:
Nit: I think it would help reader to understand that resources not include are considered deleted.

```suggestion
	// for some types. This is relied on by xds-grpc which is explicitly requesting clusters and listeners
	// but expects to receive all existing resources for those types. Missing clusters or listeners are 
	// considered deleted.
```